### ISSUE TITLE
Grid repetitions not averaging

### ISSUE THREAD

**Issue by ThorvaldLarsen** [zhinst/zhinst-qcodes#45]
I am trying to run repetitions for averaging on the DAQ module for reduced noise in measurements however the averaging is not being applied. Setting repetitions will increase the measurement time as expected but the data being fed back is not actually being averaged compared to a run with repetitions set to 1.

I suspect the issue is that the grid operations mode is set to replace instead of averaging as shown in the screenshot from the GUI. However, this parameter node does not seem to be available in the qcodes driver? Is there anywhere else this parameter is set from qcodes python terminal?

This is running with version 0.3.4 as pulled by pip install.

![image](https://user-images.githubusercontent.com/23334244/178827484-7b5c39b5-76c9-4b93-9e0b-3fd69c0b925a.png)

![image](https://user-images.githubusercontent.com/23334244/178827373-abfc5a92-98d7-41e2-960c-42d04d0f8e3e.png)

**Comment by tobiasah** [comment]
@ThorvaldLarsen thanks for bringing this up. Since the qcodes driver had a big refactoring earlier this year we are still lacking some documentation (but we are working on that). Therefor the LabOne documentation is often a good reference point. 

Also the LabOne UI is a good starting point. Don't know if you know this feature but there is a so called `API Log` which prints out the corresponding Python API commands needed to do the same measurement in Python. Although the are for the native Python API of LabOne they can easily be translated to our QCoDeS driver (Since the nodes are the same, with the difference of not beeing strings but objects). 

For your particular problem the LabOne DAQ Module documentation is a good reference (https://docs.zhinst.com/labone_programming_manual/data_acquisition_module.html#pm.core.modules.daq.signalsubscription). 

To summarize it: 
If averaging is activated (grid/repetitions > 1) you must use the `.avg` suffix, e.g. `/DEV…./DEMODS/N/SAMPLE.r.avg` when subscribing to your nodes in the daq module. The UI does this automatically for you and offers the handy drop down menu which does not translate directly to any node (if it would the tool tips would show the corresponding node). 

I hope the clarifies the issue. If not feel free to ask some follow up questions. 

PS: The one of our DAQ Examples from the LabOne Python API is using this as well hands on (https://github.com/zhinst/labone-api-examples/blob/d4be8b1d3ab2ad653a9669cdaf1bae12dad98d43/common/python/example_data_acquisition_edge.py#L215) maybe this helps as well.

**Comment by ThorvaldLarsen** [comment]
Thanks for the quick reply and links to references. Can you comment on what changed for this behavior since the refactoring of 0.3? When running with 0.2 version the averaging was working with the standard ".x" subscription to pull the data. Simple setting the repetition would automatically make the returned data be averaged.

**Comment by tobiasah** [comment]
Between 0.2.x and 0.3.0 we basically rewrote both our qcodes driver and the underlying zhinst-toolkit. The benefits are a significant speed improvement and more important the availability of all nodes of any device from Zurich Instruments (they are no longer hard coded but read from the device on every connect). In addition both in the qcodes driver and zhinst-toolkit a lot of additional logic (probably also the feature you describe) where removed since it either did not work in all cases or prevented edge cases. 

Basically everything works now the same way than in LabOne with the convenience of some static helper functions and the easy node access through arguments/items.

I know this is inconvenient for existing users but it brings the benefit of full feature support, maintainability and stability.

PS: The 0.2.x driver will continue to work even with newer LabOne releases for a while (Although we will not actively maintain them)

**Comment by ThorvaldLarsen** [comment]
Thanks for the clarification on the different versions. Closing this issue as I just need to update measurement functions to match the changes for 0.3.
